common/lib/dependabot/sem_version.rb

```diff
@@ -0,0 +1,134 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+# See https://semver.org/spec/v2.0.0.html for semver 2 details
+#
+module Dependabot
+  class SemVersion
```

Comment:
maybe rename and call this SemVer2 for clarity?

Reply:
I can do that.